internal/config/variables/variables.go

```diff
@@ -666,8 +680,71 @@ func EvaluateVariables(
 			})
 		}
 	}
+	// Error here if we have them from parsing
+	if diags.HasErrors() {
+		return nil, nil, diags
+	}
+
+	jobVals, diags := getJobValues(vs, iv, salt)
+	if diags.HasErrors() {
+		return nil, nil, diags
+	}
+	return iv, jobVals, diags
+}
+
+// getJobValues combines the Variable and Value into a VariableRef,
+// encoding any 'sensitive' values as SHA256 values with the given salt.
+func getJobValues(vs map[string]*Variable, values Values, salt string) (map[string]*pb.Variable_Ref, hcl.Diagnostics) {
+	varRefs := make(map[string]*pb.Variable_Ref, len(values))
+	var diags hcl.Diagnostics
+	for v, value := range values {
+		var val string
+		var t string
+		switch value.Value.Type() {
+		case cty.String:
+			val = value.Value.AsString()
+			t = "string"
+		case cty.Bool:
+			b := value.Value.True()
+			val = fmt.Sprintf("%t", b)
+			t = "bool"
+		case cty.Number:
+			var num int64
+			err := gocty.FromCtyValue(value.Value, &num)
+			if err != nil {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Invalid number",
+					Detail:   err.Error(),
+				})
+				return nil, diags
+			}
+			val = fmt.Sprintf("%d", num)
+			t = "int"
+		default:
+			// handle any HCL complex types
+			bv := hclwrite.TokensForValue(value.Value).Bytes()
+			buf := bytes.NewBuffer(bv)
+			val = buf.String()
+			t = "complex"
+		}
+
+		if vs[v].Sensitive {
+			// salt shaker
+			saltedVal := salt + val
+			h := sha256.Sum256([]byte(saltedVal))
+			hVal := h[:]
+			val = hex.EncodeToString(hVal)
```

Comment:
```suggestion
			val = hex.EncodeToString(h[:])
```
I'd skip defining a single-use value